oaknational/oak-ai-lesson-assistant
# feat: shortened text responses, basedOn selection and generate mode prompt tidy-up

## Description

- Adapts the prompt so that we more explicitly state that we want the user to select a lesson from the ones pulled from the database
- Removes the relevant lessons section and prompt entirely if there are no lessons found
- Attempts to skip the first "shall we start from scratch" (as per the [Miro](https://miro.com/app/board/uXjVKs4B7Eg=/) spec) but this doesn't seem to be consistent
- Attempts to reduce the text response length to meet the Miro spec by providing examples in-line with each step
- Numbers each step so that it's clearer what the happy path should look like
- Fixes a few issues with the prompt which were mentioning out of date things
- Restructures the prompt further so that interactive mode and generate mode are more separate - we were mixing up various user-interaction parts of the prompt with what would be used for non-interactive generation
- Renames some files to be clearer
- Breaks out the voice and language into a separate prompt file
- Tries to ensure that the additionalMaterials section does not get overwritten when making edits

## Issue(s)

Fixes the issue where Aila immediately picks a lesson plan to base upon.

Reduces text response length since it grew longer after merging SO

## How to test

* Try making a lesson base on an existing one. It should ask you before defaulting to one of the options.
* Check that the text output in the lesson plan meets the spec in terms of text length.
* Check that the text responses on the left hand side look okay. They are somewhat different, but shorter now.
